carbon-design-system/carbon
# fix(ComboBox): handle custom value on input blur

Closes #16864

This pull request fixes the handling of custom values on input blur. The changes ensure that the ComboBox correctly handles custom values again and updates the selected item accordingly. Also includes a refactoring to remove else-returns in the input blur switch block.

#### Changelog

**Changed**

- ComboBox input blur behavior

#### Testing / Reviewing

- visit the ComboBox story
- enable `allowCustomValue`
- select a preexisting value
- modify the selection to a custom value
- confirm that the previous selection is not highlighted when the component is reopened
- confirm that the new value remains in the component
